pulp_rpm/app/models/repository.py

```diff
@@ -440,8 +441,27 @@ class RpmDistribution(Distribution):
     def content_handler(self, path):
         """Serve config.repo and public.key."""
         if path == self.repository_config_file_name:
+            publication = None
+            if self.publication:
+                publication = self.publication.cast()
+                repository = publication.repository.cast()
+            elif self.repository:
+                repository = self.repository.cast()
+                versions = repository.versions.all()
+                publications = Publication.objects.filter(
+                    repository_version__in=versions, complete=True
+                )
+                try:
+                    publication = (
+                        publications.select_related("repository_version")
+                        .latest("repository_version", "pulp_created")
+                        .cast()
+                    )
+                except ObjectDoesNotExist:
+                    pass
+            if not publication:
```

Comment:
Could you refactor this block of code into something like

```
def find_repository_and_publication(self):
    # all of this logic above
    return (repository, publication)

(repository, publication) = self.find_repository_and_publication()
if not publication:
    return
```

I wish there was a better name for such a function, probably there is a slightly better abstraction that I haven't thought of, but it should at least avoid adding too many temporary variables into the namespace.

Another reason is that this logic possibly deserves to be available in pulpcore so that other plugins can use it.  I don't think we should worry too much about it until we figure out what the right abstractions are, but it would be good to keep this code semi-separate for that reason as well.

We can file a refactor issue for it.

Reply:
I'll look into moving the code into pulpcore (probably onto the master distribution model, that way it is present in the content app and for every plugin)